### PR TITLE
chore: removes local cache used for rotation

### LINKED ISSUE #582: Upgrade to controller-runtime `v0.9.0`
- Upgrade to controller-runtime `v0.9.0`
  - Use `SelectorsByObject` to do a filtered ListWatch
  - This will enable us to remove the vendored code for `pkg/cache`
- Investigate using `controller-runtime` for rotation reconciler

ref: https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.9.0

/assign

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**What this PR does / why we need it**:
This PR removes local informer cache used for rotation reconciler. Instead now it uses manager's cache.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #582 

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
/ok-to-test

File: pkg/rotation/reconciler.go [hunk 1]
```diff
@@ -246,20 +326,28 @@ func (r *Reconciler) reconcile(ctx context.Context, spcps *v1alpha1.SecretProvid
 	// read the Kubernetes secret referenced in NodePublishSecretRef and marshal it
 	// This comprises the secret parameter in the MountRequest to the provider
 	if nodePublishSecretRef != nil {
-		secretName := strings.TrimSpace(nodePublishSecretRef.Name)
-		secretNamespace := spcps.Namespace
+		// // read secret from the manager's cache
+		// secret := v1.Secret{}
+		// err = r.cache.Get(
+		// 	ctx,
```
Comment (aramase) [inline pkg/rotation/reconciler.go:332]: If this is unused, could you remove it?

File: pkg/rotation/reconciler.go [hunk 2]
```diff
@@ -419,8 +507,15 @@ func (r *Reconciler) updateSecretProviderClassPodStatus(ctx context.Context, spc
 
 // patchSecret patches secret with the new data and returns error if any
 func (r *Reconciler) patchSecret(ctx context.Context, name, namespace string, data map[string][]byte) error {
-	secret := &v1.Secret{}
-	secret, err := r.store.GetSecret(name, namespace)
+	secret := v1.Secret{}
```
Comment (aramase) [inline pkg/rotation/reconciler.go:510]:
```suggestion
	secret := &v1.Secret{}
```

File: pkg/rotation/reconciler.go [hunk 3]
```diff
@@ -159,6 +161,71 @@ func (r *Reconciler) Run(stopCh <-chan struct{}) {
...
+	ctx := context.Background()
+	var err error
+
+	key, quit := r.queue.Get()
+	if quit {
+		return false
+	}
+	defer r.queue.Done(key)
+
+	spcps := v1alpha1.SecretProviderClassPodStatus{}
```
Comment (aramase) [inline pkg/rotation/reconciler.go:182]:
```suggestion
	spcps := &v1alpha1.SecretProviderClassPodStatus{}
```

File: docs/book/src/topics/secret-auto-rotation.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ Depending on how the application consumes the secret data:
...
+To enable auto rotation, enable the `--enable-secret-rotation` feature gate for the `secrets-store` container in the Secrets Store CSI Driver pods. The rotation poll interval can be configured using `--rotation-poll-interval`. The default rotation poll interval is `2m`. If using helm to install the driver, set `enableSecretRotation: true` and configure the rotation poll interval by setting `rotationPollInterval`. The rotation poll interval can be tuned based on how frequently the mounted contents for all pods and Kubernetes secrets need to be resynced to the latest. _Do not_ specify `objectVersion` in SecretProviderClass.
```
[the remark this answers is not in the document]
Comment (nilekhc) [reply docs/book/src/topics/secret-auto-rotation.md:19]: Sounds right.

File: pkg/rotation/reconciler.go [hunk 5]
```diff
@@ -73,42 +74,40 @@ const (
...
 	providerVolumePath   string
-	scheme               *runtime.Scheme
 	rotationPollInterval time.Duration
 	providerClients      *secretsstore.PluginClientBuilder
 	queue                workqueue.RateLimitingInterface
 	reporter             StatsReporter
 	eventRecorder        record.EventRecorder
 	kubeClient           kubernetes.Interface
 	crdClient            versioned.Interface
+	cache                client.Reader
```
[the remark this answers is not in the document]
Comment (nilekhc) [reply pkg/rotation/reconciler.go:85]: @aramase The cache will contains all K8s resources plus CRDs. Do you mean to document selectors we are using to populate certain resources?

**Review by aramase — COMMENTED** [review]
@nilekhc The changes look good! We should also run load tests with the similar cluster setup as https://secrets-store-csi-driver.sigs.k8s.io/load-tests.html to document the memory consumption.

**Comment by nilekhc** [comment]
Load test result for top 5 CSI pods (without pprof enabled):
```
NAME                      CPU(cores)   MEMORY(bytes)   
csi-secrets-store-p2f92   2m           55Mi            
csi-secrets-store-9z7nm   2m           55Mi            
csi-secrets-store-k57d4   1m           54Mi            
csi-secrets-store-hmqzs   2m           54Mi            
csi-secrets-store-4lp76   2m           54Mi            
```

Load test result for top 5 CSI pods (with pprof enabled):
```
NAME                      CPU(cores)   MEMORY(bytes)
csi-secrets-store-54sxr   2m           59Mi
csi-secrets-store-hhqc5   2m           59Mi
csi-secrets-store-kx77t   2m           59Mi
csi-secrets-store-994xs   2m           58Mi
csi-secrets-store-fm25p   2m           58Mi
```

**Review by aramase — COMMENTED** [review]
/approve

**Pull request merged**